### DISCUSSION TITLE
Resident Size in Flamegraph

### DISCUSSION THREAD

**Discussion by Makhbaz** [bloomberg/memray#527] [Q&A] [answered]
Hi all.
I'm trying to find out memory leaks in an application and here is the graph:
![newplot](https://github.com/bloomberg/memray/assets/109301017/e602bc2b-ed0e-4db0-a48c-e621fce09029)
![image](https://github.com/bloomberg/memray/assets/109301017/7a6811fc-28f8-47c7-9b5c-a6864524fc08)
I used `--leaks` to ensure it's at the end of the track.
The resident size is growing, but I can only see the heap size here.
Is there any way to process the resident size?

**Comment by pablogsal** [comment, marked as answer]
Unofortunately there is no way to map resident size to specific allocations because what makes resident size grow or shrink is detached from the allocators themselves. For example, resident size can grow when the OS actually has to allocate memory that has been requested via `mmap` but can suddenly shrink for no apparent reason because pages are being swaped out. You can find more information about this in https://bloomberg.github.io/memray/memory.html

**Reply by Makhbaz** [reply]
Thanks
